x/virtualgroup/keeper/keeper.go

```diff
@@ -0,0 +1,684 @@
+package keeper
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+
+	"github.com/bnb-chain/greenfield/internal/sequence"
+	types2 "github.com/bnb-chain/greenfield/types"
+	sptypes "github.com/bnb-chain/greenfield/x/sp/types"
+	storagetypes "github.com/bnb-chain/greenfield/x/storage/types"
+	"github.com/bnb-chain/greenfield/x/virtualgroup/types"
+)
+
+type (
+	Keeper struct {
+		cdc       codec.BinaryCodec
+		storeKey  storetypes.StoreKey
+		tStoreKey storetypes.StoreKey
+		authority string
+
+		// Keepers
+		spKeeper      types.SpKeeper
+		accountKeeper types.AccountKeeper
+		bankKeeper    types.BankKeeper
+		paymentKeeper types.PaymentKeeper
+		// sequence
+		lvgSequence       sequence.Sequence[uint32]
```

Comment:
Do we need a global  `lvgSequence`? To my understanding, `LVG` is bucket scope.

Reply:
If not global, should we maintain a sequence for every bucket?